app/client/src/store/mutations.ts

```diff
@@ -50,11 +50,19 @@ export default {
   },
   addMicroreactURL(state: RootState, URLinfo: Record<string, string>) {
     state.results.perCluster[URLinfo.cluster] = {
+      ...state.results.perCluster[URLinfo.cluster],
       cluster: URLinfo.cluster,
       microreactURL: URLinfo.url,
     };
   },
   setToken(state: RootState, token: string | null) {
     state.microreactToken = token;
   },
+  addGraphml(state: RootState, GraphInfo: Record<string, string>) {
+    state.results.perCluster[GraphInfo.cluster] = {
+      ...state.results.perCluster[GraphInfo.cluster],
```

Comment:
Why do you need the spread operator here? What might be retained from any previous graphInfo for the cluster?

Reply:
I did this to prevent that Microreact URLs which get also stored in `state.results.perCluster[clusternumber]` are overwritten/ deleted. Also did the same the other way round in `addMicroreactURL()`, to prevent the graph data being deleted when already stored.